integ-test/src/test/java/org/opensearch/sql/ppl/DateTimeImplementationIT.java

```diff
@@ -0,0 +1,126 @@
+            TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2008-12-24 21:30:00"));
+  }
+
+  @Test
+  public void inRangeZeroToPositive() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("source=%s | eval f = DATETIME('2008-12-25 05:30:00+00:00', '+01:00') | fields f",
+            TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2008-12-25 06:30:00"));
+  }
+
+  @Test
+  public void inRangeNegativeToPositive() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("source=%s | eval f = DATETIME('2008-12-25 05:30:00-05:00', '+05:00') | fields f",
+            TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2008-12-25 15:30:00"));
+  }
+
+  @Test
+  public void inRangeTwentyHourOffset() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("source=%s | eval f = DATETIME('2004-02-28 23:00:00-10:00', '+10:00') | fields f",
+            TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2004-02-29 19:00:00"));
+  }
+
+
+  @Test
+  public void inRangeYearChange() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("source=%s | eval f = DATETIME('2008-01-01 02:00:00+10:00', '-10:00') | fields f",
+            TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2007-12-31 06:00:00"));
+  }
+
+  @Test
+  public void inRangeZeroToMax() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("source=%s | eval f = DATETIME('2008-12-25 05:30:00+00:00', '+14:00') | fields f",
+            TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2008-12-25 19:30:00"));
+  }
+
+  @Test
+  public void inRangeNoToTZ() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("source=%s | eval f = DATETIME('2008-01-01 02:00:00+10:00') | fields f", TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2008-01-01 02:00:00"));
+  }
+
+  @Test
+  public void inRangeNoTZ() throws IOException {
+    JSONObject result = executeQuery(
+        String.format("source=%s | eval f = DATETIME('2008-01-01 02:00:00') | fields f", TEST_INDEX_DATE));
+    verifySchema(result, schema("f", null, "datetime"));
+    verifySome(result.getJSONArray("datarows"), rows("2008-01-01 02:00:00"));
+  }
+
+  @Test
+  public void nullFromFieldOver() throws IOException {
```

Comment:
Is this `Input1`?

Reply:
Yes. Should it be more verbose? `From` is that TZ field name in the documentation.